lib/assent/strategies/slack.ex

```diff
@@ -32,40 +32,27 @@ defmodule Assent.Strategy.Slack do
   Instead you should set `team: TEAM_ID` in the `authorization_params` keyword
   list.
   """
-  use Assent.Strategy.OAuth2.Base
+  use Assent.Strategy.OIDC.Base
 
   alias Assent.Config
 
   @impl true
   def default_config(config) do
     [
       site: "https://slack.com",
-      token_url: "/api/oauth.access",
-      user_url: "/api/users.identity",
       authorization_params: authorization_params(config),
-      auth_method: :client_secret_post
+      client_authentication_method: "client_secret_post"
     ]
   end
 
   defp authorization_params(config) do
-    default = [scope: "identity.basic identity.email identity.avatar identity.team"]
+    default = [scope: "openid email profile"]
     case Config.fetch(config, :team_id) do
       {:ok, team_id} -> Config.put(default, :team, team_id)
       _error         -> default
     end
   end
 
   @impl true
-  def normalize(_config, identity) do
-    {:ok, %{
-      "sub"                => uid(identity),
-      "name"               => identity["user"]["name"],
-      "picture"            => identity["user"]["image_48"],
-      "email"              => identity["user"]["email"]
-    }, %{
-      "slack_team" => identity["team"]
-    }}
-  end
-
-  defp uid(%{"user" => %{"id" => id}, "team" => %{"id" => team_id}}), do: "#{id}-#{team_id}"
```

Comment:
It's not nice that we don't get the `team_id` anymore.
I was checking the callbacks and found `fetch_user/2`. Couldn't we do sth like that to get the `team_id`?

``` elixir
@impl true
def fetch_user(config, token) do
    with {:ok, user} <- OIDC.fetch_user(config, token),
         {:ok, jwt} <- OIDC.validate_id_token(config, token["id_token"]) do
      {:ok, Map.merge(user, %{"team_id" => jwt.claims["https://slack.com/team_id"]})}
    end
end
```